release-tools/changelog.py

```diff
@@ -40,27 +40,47 @@ class Distro(NamedTuple):
 ]
 
 
-def rewrite_version_number_file(file, pattern, version):
-    # simple sed implementation, read all the lines first, then write them out
-    # again, applying the given pattern to every line (the pattern is expected
-    # to only ever match one line)
-    with open(file, "r") as fh:
-        lines = fh.readlines()
-    with open(file, "w") as fh:
-        for line in lines:
-            fh.write(re.sub(pattern + ".+$", pattern + version, line))
+def rewrite_version_number_file(file_name, pattern, version, write):
+    """rewrite_version_number_file reads a packaging file with a version number
+    in it and updates the version number to the new one specified using the
+    regex pattern.
+    It returns the new file contents and optionally writes out the new contents
+    back to the file as well.
+    """
+
+    with open(file_name, "r") as fh:
+        file_bytes = fh.read()
```

Comment:
FWIW, we're opening file in text mote ("r" vs "rb"), so we'll have a string here, so `*_bytes` is confusing both here and later occurrences (like new_bytes)

Reply:
Good point, I changed these variables